pkg/replication/replication.go

```diff
@@ -266,6 +272,10 @@ func (r replication) handleRenewalErrors(session kp.Session, renewalErrCh chan e
 	select {
 	case <-r.replicationDoneCh:
 	case <-r.replicationCancelledCh:
+		// If the replication is enacted, wait for it to exit
+		select {
+		case <-r.enactedCh:
```

Comment:
r.quitCh will not close until the select block is done because it is being deferred.